lib/puma/io_buffer.rb

```diff
@@ -1,11 +1,36 @@
 # frozen_string_literal: true
 
+require 'stringio'
+
 module Puma
-  class IOBuffer < String
-    def append(*args)
-      args.each { |a| concat(a) }
+  class IOBuffer < StringIO
```

Comment:
Would there be any benefit in using Ruby’s `IO::Buffer` if it’s available?

Reply:
I looked at the API, and it's a bit more 'low level' than `StringIO`.  I switched to StringIO because it's more 'byte' oriented (or encoding agnostic), and it works reasonably well with both String & IO objects.

I haven't benched anything, but I normally use Ruby master locally, so I suppose...